File: kravets_roman/bridge/Square.cs
using System;

namespace Shapes
{
    class Square : Shape
    {
        private double side;

        public Square(double side, IColor color) : base(color)
        {
            this.side = side;
        }

        public override void Draw()
        {
            Console.WriteLine($"Drawing a {color.Fill()} square with side {side}");
        }

        public override double CalculateArea()
        {
            return side * side;
        }

        public override double CalculatePerimeter()
        {
            return 4 * side;
        }
    }
}

File: bryniarskyi_nazarii/bridge/shape/impl/Square.java
package org.example.bridge.shape.impl;

import org.example.bridge.color.Color;
import org.example.bridge.shape.Shape;

public class Square extends Rectangle {

    private final double side;

    public Square(Color color, double side) {
        super(color, side, side);
        this.side = side;
    }
}
